src/taco-utils/promisesUtils.ts

```diff
@@ -26,12 +30,95 @@ module TacoUtility {
         public static wrapExecution<T>(initializer: () => void, codeToExecute: () => T,
-            finalizer: () => void, failHandler: (reason: any) => void = () => { /* Do Nothing */ } ): T {
+            finalizer: () => void, failHandler: (reason: any) => void = () => { /* Do Nothing */ }): T {
             initializer();
             var result = codeToExecute();
             Q(result).finally(finalizer).fail(failHandler);
             return result;
         }
+
+        /**
+         * Sequentially runs a number of promises obtained from an array of values
+         *  @value: array of values used to create chain of promises
+         *  @func: accumulator function which runs over each array value and returns a promise which resolves to an accumulated value
+         *  @initialValue: initial accumulated value
+         */
+        public static chain<T, U>(values: T[], func: (value: T, valueSoFar: U) => Q.Promise<U>, initialValue?: U): Q.Promise<U> {
+            return values.reduce(function(soFar: Q.Promise<U>, val: T): Q.Promise<U> {
+                return soFar.then(function(valueSoFar: U): Q.Promise<U> {
+                    return func(val, valueSoFar);
+                });
+            }, Q(initialValue));
+        }
+
+        /**
+         * Syntactic sugar for promise or
+         *  @conditions: a variable array of promises which resolve to true/false
+         *
+         *  @returns: resolves to true if either of the promises resolve to true, false otherwise
+         */
+        public static or(...conditions: PromiseFuncOrValue<boolean>[]): Q.Promise<boolean> {
+            return PromisesUtils.logicalOp(true, ...conditions);
+        }
+
+        /**
+         * Syntactic sugar for promise and
+         *  @conditions: a variable array of promises which resolve to true/false
+         *
+         *  @returns: resolves to true if both the promises resolve to true, false otherwise
+         */
+        public static and(...conditions: PromiseFuncOrValue<boolean>[]): Q.Promise<boolean> {
+            return PromisesUtils.logicalOp(false, ...conditions);
+        }
+
+        /**
+         * Syntactic sugar for if/else style promises
+         *  @condition: Promise resolving to true/false 
+         *  @promiseTrue: then promise if condition resolves to "true"
+         *  @promiseFalse: then promise if condition resolves to "false"
+         */
+        public static condition<T>(condition: Q.Promise<boolean> | boolean,
+            promiseTrue: (() => Q.Promise<T>) | T,
+            promiseFalse: (() => Q.Promise<T>) | T): Q.Promise<T> {
+
+            var _promiseTrue: () => Q.Promise<T>  = PromisesUtils.getPromiseFunc(promiseTrue);
+            var _promiseFalse: () => Q.Promise<T> = PromisesUtils.getPromiseFunc(promiseFalse);
+
+            if (typeof condition === "boolean") {
+                return condition ? _promiseTrue() : _promiseFalse();
+            }
+            return (<Q.Promise<boolean>>condition)
```

Comment:
Alternately, you could just always wrap it, collapsing both cases together with something like `return Q(condition).then(function(result: Boolean): ...`